pages/LivePage.tsx

```diff
@@ -27,9 +33,146 @@ export function renderSection(
   );
 }
 
-export default function LivePage({ sections }: Props) {
+interface SlotImplementation {
+  slot: PreactComponent<JSX.Element, UseSlotProps>;
+  used: boolean;
+}
+
+/**
+ * Builds a map which the key is the name of the slot and the value is the slot component itself.
+ * For those sections that aren't used inside a slot it is considered the default `content slot`.
+ * @param sections the sections
+ * @returns the implementation map.
+ */
+function indexedImplementations(
+  sections: Section[],
+) {
+  const useContentSlot: Record<string, SlotImplementation> = {
+    content: {
+      slot: {
+        Component: UseSlot,
+        props: {
+          name: "content",
+          sections: [] as Section[],
+        },
+        metadata: {
+          component: "$live/sections/UseSlot.tsx",
+          resolveChain: [],
+        },
+      },
+      used: false,
+    },
+  };
+  return sections
+    ?.reduce(
+      (indexed, sec) => {
+        if (isSection(sec, "$live/sections/UseSlot.tsx")) {
+          return {
+            ...indexed,
+            [sec.props.name]: {
+              slot: sec,
+              used: false,
+            },
+          };
```

Comment:
wdyt about mutating the reducer acculator, like this:

```ts
indexed[sec.props.name] = { slot: sec, used: false };
```

The same suggestion works for the code block outside this if statement